NewDawn/NewDawn/ViewControllers/LoginViewControllers/PhoneAuthenticateViewController.swift

```diff
@@ -133,11 +134,36 @@ class PhoneAuthenticateViewController: UIViewController {
             _ = LoginUserUtil.saveAccessToken(token: "N/A")
             // Phone verification success. Go to main registration page
             // Go to profile gender/name/birthday fill page
-            DispatchQueue.main.async {
-                let mainPageStoryboard:UIStoryboard = UIStoryboard(name: "MainPage", bundle: nil)
-                let homePage = mainPageStoryboard.instantiateViewController(withIdentifier: "MainTabViewController") as! MainPageTabBarViewController
-                let appDelegate = UIApplication.shared.delegate
-                appDelegate?.window??.rootViewController = homePage
+            
+            if LoginUserUtil.isLogin() {
+                LoginUserUtil.fetchLoginUserProfile(readLocal: false) {
+                    user_profile, error in
+                    if error != nil {
+                        self.displayMessage(userMessage: "Error: Fetch Login User Profile Failed for user id \(String(describing: LoginUserUtil.getLoginUserId())): \(error!). This can happen if you don't have accessToken stored locally")
+                        return
+                    }
+                    if user_profile != nil {
+                        let queue = TaskQueue()
+                        queue.tasks +=~ {
+                            LoginUserUtil.downloadOverwriteLocalInfo(profile: user_profile!)
+                        }
+                        queue.tasks +=~ {
+                            LoginUserUtil.downloadOverwriteLocalImages(profile: user_profile!)
+                        }
+                        queue.tasks +=! {
+                            let mainPageStoryboard:UIStoryboard = UIStoryboard(name: "MainPage", bundle: nil)
+                            let homePage = mainPageStoryboard.instantiateViewController(withIdentifier: "MainTabViewController") as! MainPageTabBarViewController
+                            let appDelegate = UIApplication.shared.delegate
+                            appDelegate?.window??.rootViewController = homePage
+                        }
+                        queue.run()
```

Comment:
Damn this is super awesome. Is TaskQueue() a built-in in Swift?

Reply:
HAHAHAH. No, I thought you installed this library.

Reply:
Haha guess this was installed as a dependency of other libraries.
How did you find this though?

Reply:
When I was installing MixPanel, I found TaskQueue in the "frameworks" folder, so I looked it up. It turns out to be a convenient library to use.